### ISSUE TITLE
nnn fails to install on darwin (MacOS) because of archivemount dependency

### ISSUE THREAD

**Issue by mamciek** [NixOS/nixpkgs#119728]
**Describe the bug**
nnn fails to install on MacOs (darwin) because archivemount is not supported on darwin platform

**To Reproduce**
1. nix-shell -p nnn
```
error: Package ‘archivemount-0.9.1’ in /nix/store/9xg5s5yk4kjw6f00ps0ync2xq56n4866-nixpkgs-21.05pre282987.294d1925af6/nixpkgs/pkgs/tools/filesystems/archivemount/default.nix:18 is not supported on ‘x86_64-darwin’, refusing to evaluate.
```

**Notify maintainers**
@jfrankenau @Br1ght0ne @sikmir @tomberek @siraben

**Metadata**
Please run `nix-shell -p nix-info --run "nix-info -m"` and paste the result.
```
 - system: `"x86_64-darwin"`
 - host os: `Darwin 20.3.0, macOS 10.16`
 - multi-user?: `no`
 - sandbox: `no`
 - version: `nix-env (Nix) 2.3.10`
 - channels(maciej.mazur): `"home-manager, nixpkgs-21.05pre282987.294d1925af6"`
 - nixpkgs: `/Users/maciej.mazur/.nix-defexpr/channels/nixpkgs`
 ```

Maintainer information:
```yaml
# a list of nixos modules affected by the problem
module:  nixpkgs/pkgs/applications/misc/nnn
```

**Comment by midchildan** [comment]
This is fixed in #118269.

<details>

```console
❯ nix-build -A nnn
these derivations will be built:
  /nix/store/6rwg101ybsgzvhiimrps0jzhvkk41yxi-nnn-3.6.drv
these paths will be fetched (0.15 MiB download, 0.64 MiB unpacked):
  /nix/store/ph6zv6q89acfwxvbf2fm45sh2m40zxsx-source
copying path '/nix/store/ph6zv6q89acfwxvbf2fm45sh2m40zxsx-source' from 'https://cache.nixos.org'...
building '/nix/store/6rwg101ybsgzvhiimrps0jzhvkk41yxi-nnn-3.6.drv'...
unpacking sources
unpacking source archive /nix/store/ph6zv6q89acfwxvbf2fm45sh2m40zxsx-source
source root is source
patching sources
configuring
no configure script, doing nothing
building
build flags: SHELL=/nix/store/4k9jawhj2d4jzh6n1p5qk63rrdnzxv48-bash-4.4-p23/bin/bash PREFIX=\$\(out\)
clang  -std=c11 -Wall -Wextra -Wshadow -O3 -D_DARWIN_C_SOURCE -I/nix/store/f7kn2avlfsn992klwywb2yjf9w2bsgzf-ncurses-6.2-dev/include  -o nnn src/nnn.c -lreadline -Wl,-search_paths_first -lncursesw
installing
install flags: SHELL=/nix/store/4k9jawhj2d4jzh6n1p5qk63rrdnzxv48-bash-4.4-p23/bin/bash PREFIX=\$\(out\) install
install -m 0755 -d /nix/store/v5zv928jr54yfvsxqvkw1hd9lrkq89qg-nnn-3.6/bin
install -m 0755 nnn /nix/store/v5zv928jr54yfvsxqvkw1hd9lrkq89qg-nnn-3.6/bin
install -m 0755 -d /nix/store/v5zv928jr54yfvsxqvkw1hd9lrkq89qg-nnn-3.6/share/man/man1
install -m 0644 nnn.1 /nix/store/v5zv928jr54yfvsxqvkw1hd9lrkq89qg-nnn-3.6/share/man/man1
post-installation fixup
gzipping man pages under /nix/store/v5zv928jr54yfvsxqvkw1hd9lrkq89qg-nnn-3.6/share/man/
strip is /nix/store/z8d39gw4q90i97jlsilbad33siwh4yxd-cctools-binutils-darwin-949.0.1/bin/strip
stripping (with command strip and flags -S) in /nix/store/v5zv928jr54yfvsxqvkw1hd9lrkq89qg-nnn-3.6/bin
patching script interpreter paths in /nix/store/v5zv928jr54yfvsxqvkw1hd9lrkq89qg-nnn-3.6
/nix/store/v5zv928jr54yfvsxqvkw1hd9lrkq89qg-nnn-3.6
```
</details>

**Comment by midchildan** [comment]
Actually, that wasn't enough. I overlooked that archivemount was only a part of [recently introduced dependencies](https://github.com/NixOS/nixpkgs/commit/5c3a1c28086b8d4f1443fb39d6c4bd26e2e2c7a6). For the latest nnn to build on Darwin, the vlock dependency would have to be disabled.

Moreover, the dependencies that were added are command line utilities that nnn can optionally use if installed. As these are command line utilities that users would typically choose to explicitly install if needed, I think it's enough to let nnn pick up the available commands from `PATH` at runtime rather than having the nnn package depend on them. Or maybe introduce an option (e.g., `withExtraTools`) to include those dependencies.
